pkgs/development/python-modules/scipy/default.nix

```diff
@@ -17,46 +22,156 @@
 , pybind11
 , pooch
 , libxcrypt
+, xsimd
+, blas
+, lapack
+# These 2 are used at scipy/meson.build - if you want to override the blas
+# implementation, you'll likely only need to override blas and lapack deps, but
+# this might be needed in some cases.
+, blasName ? blas.pname
```

Comment:
> Do we _have_ to expose these two parameters? If we have not needed it so far, then let's not. It can be added in case someone encounters the issue. This would probably be something for our blas wrapper to deal with.

Now that we use `mesonFlags`, perhaps we can hope that users will use something like:

```nix
myScipy = super.scipy.overridePythonAttrs(old: {
  mesonFlags = old.mesonFlags ++ [
    "-Dblas="mkl_blas" # or whatever
    # ...
  ];
});
```

Does that sound good for you @SomeoneSerge ?

Reply:
The manual documents a way to enable MKL at the package set level: https://nixos.org/manual/nixpkgs/unstable/#sec-overlays-alternatives-blas-lapack. Scipy is an important building block, I think we better ensure it's compliant with the documentation

Reply:
@SomeoneSerge I documented the blas/lapack overriding nix code for Python. Your review is also welcome.

Reply:
And I removed these `blasName ? blas.pname` arguments, as indeed they shouldn't be needed - unless someone is doing something really wrong. Hence I'm marking this conversation as resolved.